cpp/src/arrow/dataset/scanner_test.cc

```diff
@@ -1503,6 +1518,7 @@ TEST(ScanNode, Schema) {
   fields.push_back(field("__fragment_index", int32()));
   fields.push_back(field("__batch_index", int32()));
   fields.push_back(field("__last_in_fragment", boolean()));
+  fields.push_back(field("__filename", utf8()));
```

Comment:
I don't see any test checking that we do actually get the right values in this new field

Reply:
Added a test in PyArrow which checks whether the queried augmented fields are returned. We can add another test to check the correct values are returned for the field.

Reply:
Is it worth checking the values too? (We could sort the table before comparison?)

Reply:
Though I suppose `__filename` will have a different value on every run so that may be annoying

Reply:
I have added a test which actually checks the values of the augmented fields, does the approach looks good, or can we have a better implementation.